### PR TITLE
Added Tutorial changes to reflect virtual device changes

### PR DESCRIPTION
**Pull request by shih1**
Fixed issue #193 
Changed tutorial to reflect changes. 
Added 'scan' to the tutorial.

### CODE REVIEW

File: doc/source/tutorials/streaming-data.rst [hunk 1]
```diff
@@ -172,11 +172,20 @@ Just like in the first tutorial, create a class for the virtual device::
 
             self.stream_realtime(0x1000, random.randint(32, 100))
 
-Save your device file as `demo_streamer.py` and then connect to it using the
-iotile tool::
+Save your device file as `demo_streamer.py`. 
+
+This time we'll scan for the device before connecting to it. Scanning in real life will display all ofthe devices you are able to connect to, as well as the unique id (uuid) of each device. You can then connect to it using the
```
Comment (dkarchmer) [inline doc/source/tutorials/streaming-data.rst]: "ofthe" -> "of the"

File: iotilecore/test/test_hw/test_virtualadapter.py [hunk 2]
```diff
@@ -156,3 +167,11 @@ def test_realtime_tracing(tracer_hw):
 
     wrong_data = [x for x in words if (x != 'hello' and x != 'goodbye' and x != '')]
     assert len(wrong_data) == 0
+
+def test_virtual_scan(realtime_scan_hw):
+    """Make sure we can scan for virtual devices and connect directly without connect_direct
+    """
+    devices = report_hw.scan()
```
Comment (timburke) [inline iotilecore/test/test_hw/test_virtualadapter.py]: Looks like a copy/paste error.  This should be realtime_scan_hw

Comment (timburke) [inline iotilecore/test/test_hw/test_virtualadapter.py]: Note that the continuous integration tests failed on this line as well:

https://travis-ci.org/iotile/coretools/builds/234175957?utm_source=github_status&utm_medium=notification

Comment (timburke) [inline iotilecore/test/test_hw/test_virtualadapter.py]: You can just push a fix to the same branch and the PR will update and the CI will run again.  Other than that it looks good.

**Comment by shih1** [comment]
@dkarchmer  thanks for the catch
@timburke used tox and it seems good to go now. Wating for CI to finish

**Pull request merged**